### PR TITLE
Improved cover url resolving

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-536

#### Description

This PR makes sure that we have a higher probability for getting a cover url in various places in our applications.
* Several pids are being sent at the same time to the cover service
* If we can find the best representation cover we'll use that otherwise we try to pick the first available cover

### CODE REVIEW

File: src/core/dbc-gateway/generated/graphql.tsx [hunk 1]
```diff
@@ -76,6 +76,8 @@ export enum AccessUrlType {
 
 export type Audience = {
   __typename?: "Audience";
+  /** PEGI age rating for games  */
+  PEGI?: Maybe<Pegi>;
```
Comment (Adamik10) [inline src/core/dbc-gateway/generated/graphql.tsx:80]: Is this just an updated version of the API, or do we now get PEGI for a specific reason?

File: src/components/cover/helper.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+}: {
+  coverData: CoverType[] | null | undefined;
+  bestRepresentation?: Manifestation;
+  size: CoverServiceSizes;
+}) => {
+  if (!coverData) {
+    return null;
+  }
+
+  // Make sure we only have covers in our data that has an url in the given size.
```
Comment (Adamik10) [inline src/components/cover/helper.ts:43]: Small typo: has an url -> has a url

File: src/components/cover/helper.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  if (bestRepresentationCover && getUrl(bestRepresentationCover, size)) {
+    return getUrl(bestRepresentationCover, size);
+  }
+
+  // If the best representation method failed we try the first cover.
+  if (firstCover && getUrl(firstCover, size)) {
+    return getUrl(firstCover, size);
+  }
+
+  // Everything else failed. We don't know what to do 🤷.
```
Comment (Adamik10) [inline src/components/cover/helper.ts:71]: Hahahaha love this.

**Review by Adamik10 — APPROVED** [review]
Very well done! I have a small question in regards to some updated generated files, and a tiny typo, but otherwise it's cleared from here :)

File: src/core/dbc-gateway/generated/graphql.tsx (hunk 1 above)
Comment (spaceo) [reply src/core/dbc-gateway/generated/graphql.tsx:80]: Yes, it is whatever is coming from the FBI graphql schema. I change a query and needed to regenerate the code.

File: src/components/cover/helper.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  return null;
+};
+
+export default {};
+
+// ************** VITEST ***************
```
[the remark this answers is not in the document]
Comment (spaceo) [reply src/components/cover/helper.ts:77]: 🙂

**Pull request merged**